rootfs/lua/avaya_validate_path.lua

```diff
@@ -0,0 +1,43 @@
+
+local _M = {}
+
+-- alternatively: local lrucache = require "resty.lrucache.pureffi"
+local lrucache = require "resty.lrucache"
+local cache_ttl_seconds = 300 -- five minutes
+
+-- we need to initialize the cache on the lua module level so that
+-- it can be shared by all the requests served by each nginx worker process:
+local c = lrucache.new(10000)  -- allow up to 10k items in the cache
+if not c then
+    return error("failed to create the cache: " .. (err or "unknown"))
+end
+
+
+function _M.go()
+    local is_authorized = c:get(ngx.var.request_uri)
+    -- is_authorized will be nil if it's not found in cache
+    if is_authorized then
+        return
+    elseif is_authorized == false then
+        ngx.status = ngx.HTTP_FORBIDDEN
+        ngx.say('') -- return an empty response instead of nginx default 403 HTMl response
+        ngx.exit(ngx.HTTP_FORBIDDEN)
+        return
+    end
+
+    res = ngx.location.capture('/v1/blocked-paths/validate?path=' .. ngx.var.request_uri, {share_all_vars = true, method = ngx.HTTP_GET})
```

Comment:
What is the timeout for this request ?

Reply:
It uses the `http` directive default timeout which is 60 seconds.

Reply:
what would be the behaviour ,
- If the request is timed out
- if 'v1/blocked-paths/validate' is not reachable

Reply:
For both cases, the request won't go through.
The way I see it we have two options:

allow the request if the response code != 403 and possibly have unauthorized users accessing resources
block it if response code != 200 for all users

thoughts?

Reply:
i.e. it may block valid requests as well.  Capture all these behavior in Jira/Confluence page. 

Have a closure with @mattwilliamson on the behavior. Also consider these kind of test cases along with the traffic testing.

Reply:
As this is more a product-related(allow or deny) question than a technical one, I briefly discussed it with Vignaraj and he advised on blocking it for now https://onecloud-cpaas.slack.com/archives/G01FJ58JK9B/p1628530284036100.

I'll also add Masoud as a reviewer so we can get his input too.

Reply:
Ok.  What is the error response for these cases ( timeout or account service not reachable cases etc) ?

Reply:
Error codes are returned accordingly (408 for timeout, 5xx for server related issues, etc)

Reply:
We should have a very short timeout. Maybe 5s max. If it times out, I think we should allow the request. 

Here's how to do it:
https://openresty.org/en/faq.html#can-i-set-timeout-threshold-on-subrequests

Reply:
I'll make sure to add a shorter timeout on the k8s PR